### ISSUE TITLE
Collection class properties that do not become fields in the corresponding datastore entity.

### ISSUE THREAD

**Issue by GitTom** [wovalle/fireorm#116]
I propose that there be a mechanism by which to indicate that certain properties/attributes of a @Collection class - which would normally all get written to (and updated from) the datastore - are not intended as datastore fields, so they would be ignored for all datastore operations.

Two possible ways to mark these properties come to mind...
- properties whose name begins with an underscore (the convention for indicating private which is rather redundant in TS).
- use a property decorator.

Perhaps this could also be achieved by creating a subclass of the collection class, and putting these non-datastore properties in the subclass, but I think that would be awkward and wouldn't be suitable.

My use case for this is that I want to have a property called 'status' with which I would track whether the entity had been loaded from the DS, whether the instance data was dirty and needed to be written, etc.

For the record, I first brought this up in [fireorm's gitter here](https://gitter.im/fireorm/community).

**Comment by wovalle** [comment]
I agree this would be useful. And I prefer the property decorator approach since we already use decorators to specify other attributes too!

I'll check what has been done in other projects and get back (probably in the weekend) with a proposal.

**Comment by fryertom44** [comment]
I would favour using a property decorator to mark the property as **transient**, as there could be some cases where it would be useful to save underscore-prefixed properties to the datastore.

**Comment by talesmgodois** [comment]
Get methods couldn't solve this already?

I haven't tried yest but.

```
@Collection()
export class Car {
     
   property:string;

   get transientProperty(): string{
       return "something"
   }
}
```

**Comment by danieleisenhardt** [comment]
We have encountered similar problems, or so I think. I just submitted this Pull Request for a decorator, but I did not come up with the word transient for the decorator:

https://github.com/wovalle/fireorm/pull/249

**Comment by wovalle** [comment]
fixed in #249
